src/components/modals/RequestFioAddressModal.js

```diff
@@ -0,0 +1,226 @@
+// @flow
+
+import { FormField, InputAndButtonStyle, MaterialInputStyle, Modal, ModalStyle, PrimaryButton, SecondaryButton, TertiaryButton } from 'edge-components'
+import type { EdgeCurrencyConfig } from 'edge-core-js'
+import React, { Component } from 'react'
+import { ActivityIndicator, Clipboard, Image, Text, View } from 'react-native'
+import { sprintf } from 'sprintf-js'
+
+import fioRequestsIcon from '../../assets/images/fio/SendModule_FioAddress.png'
+import s from '../../locales/strings.js'
+import styles from '../../styles/scenes/ScaneStyle'
+
+type AddressModalProps = {
+  onDone: ({ fioAddress: string, memo: string } | null) => mixed,
+  fioPlugin: EdgeCurrencyConfig,
+  isConnected: boolean
+}
+
+type AddressModalState = {
+  clipboard: string,
+  address: string,
+  addressError: string,
+  memo: string,
+  memoError: string,
+  addressValidationLoading: boolean
+}
+export class FioAddressModal extends Component<AddressModalProps, AddressModalState> {
+  fioCheckQueue: number = 0
+
+  constructor (props: AddressModalProps) {
+    super(props)
+    this.state = {
+      clipboard: '',
+      address: '',
+      addressError: '',
+      memo: '',
+      memoError: '',
+      addressValidationLoading: false
+    }
+  }
+
+  componentDidMount () {
+    this._setClipboard()
+  }
+
+  _setClipboard = async () => {
+    try {
+      const address = await Clipboard.getString()
+      const isFioAddress = await this.props.fioPlugin.otherMethods.isFioAddressValid(address)
+      if (!isFioAddress) return
+
+      this.setState({
+        clipboard: address
+      })
+    } catch (e) {
+      //
+    }
+  }
+
+  updateAddress = (address: string) => {
+    this.setState(
+      {
+        address,
+        addressError: ''
+      },
+      this.validateAddressQueue
+    )
+  }
+
+  updateMemo = (memo: string) => {
+    this.setState(
+      {
+        memo
+      },
+      this.validateMemo
+    )
+  }
+
+  onPasteFromClipboard = () => {
+    const { clipboard } = this.state
+    this.updateAddress(clipboard)
+  }
+
+  onCancel = () => {
+    this.props.onDone(null)
+  }
+
+  onSubmit = async () => {
+    const { memoError, address } = this.state
+    if (!(await this.props.fioPlugin.otherMethods.isFioAddressValid(address))) {
+      return this.setState({ addressError: s.strings.send_fio_request_error_addr_not_available })
```

Comment:
Change error to "Invalid FIO address" `fio_error_invalid_address`

Reply:
changed